client/src/legacy/GridField.js

```diff
@@ -797,67 +735,6 @@ $.entwine('ss', function($) {
         e.preventDefault();
       }
     }
-  })
-
-  /**
-   * Catch submission event in filter input fields, and submit the correct button
-   * rather than the whole form.
-   */
-  $('.grid-field .grid-field__filter-header :input').entwine({
```

Comment:
Everything related to `.grid-field__filter-header` is for the legacy filter header (i.e. `GridFieldFilterHeader::getLegacyFilterHeader()` which was removed in 5.0.0)